### ISSUE TITLE
Segmentation fault with CtrlP

### ISSUE THREAD

**Issue by balinterdi** [neovim/neovim#5206]
- `nvim --version`: 0.1.4
- Vim (version: 7.4) behaves differently? Yes, it works in vim.
- Operating system/version: OSX 10.11.5
- Terminal name/version: iTerm2 3.0.5
- `$TERM`: xterm-256color
### Actual behaviour

Invoking the :CtrlP command segfaults almost every time (it works very rarely, like once every 3 times).
### Expected behaviour

It should not segfault (and it does not in vim, see above)
### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
```

Not sure, as :CtrlP does not seem to be available when launching nvim with `nvim -u NORC`. But basically installing [CtrlP](https://github.com/ctrlpvim/ctrlp.vim) and invoking the :CtrlP command throws the segfault.

**Comment by oni-link** [comment]
Cannot reproduce on upstream and linux with the following `init.vim`
```vim
call plug#begin('~/.vim/bundle')     "Plugin manager vim-plug

Plug 'ctrlpvim/ctrlp.vim'

call plug#end()
```
What is your minimal `init.vim` that makes the segfault reproducible for you?

**Comment by balinterdi** [comment]
Thank you for looking into this.

My init.vim is symlinked to my vimrc which has the following content:

```
"load pathogen managed plugins
call pathogen#runtime_append_all_bundles()

"Use Vim settings, rather then Vi settings (much better!).
"This must be first, because it changes other options as a side effect.
set nocompatible

set encoding=utf-8

"store lots of :cmdline history
set history=100

set showcmd      "show incomplete cmds down the bottom
set showmode     "show current mode down the bottom
set shortmess=atTI "shorten vim messages to make more of them fit on a line
set cmdheight=1

set incsearch   "find the next match as we type the search
set hlsearch    "hilight searches by default
set ignorecase
set smartcase

set rnu         "relative line numbers
set number      "show line number of current line

"indent settings
set nowrap
set tabstop=2
set shiftwidth=2
set softtabstop=2
set expandtab
set list listchars=tab:\ \ ,trail:·
set autoindent

" Tab completion
set wildmode=list:longest,list:full
set wildignore+=*.o,*.obj,.git,*.rbc,*.class,.svn,coverage/*,tags

" Status bar - always have a status displayed
set laststatus=2

"folding settings
set foldmethod=indent   "fold based on indent
set foldnestmax=3       "deepest fold is 3 levels
set nofoldenable        "dont fold by default

set backspace=indent,eol,start "backspace through everything in insert mode

"set showbreak=...
"set wrap linebreak nolist

"add some line space for easy reading
set linespace=4

"disable visual bell
set visualbell t_vb=

"statusline setup
set statusline =%r%m%f\ %y
set statusline +=%=%l,%c

"turn off needless toolbar on gvim/mvim
set guioptions-=T

set formatoptions-=o "dont continue comments when pushing o/O

"load ftplugins and indent files
filetype plugin on
filetype indent on

"turn on syntax highlighting
syntax enable
syntax on

"some stuff to get the mouse going in term
set mouse=a
" set ttymouse=xterm2

"hide buffers when not displayed
set hidden

""" Key mappings

"key mapping for window navigation
map <C-h> <C-w>h
map <C-j> <C-w>j
map <C-k> <C-w>k
map <C-l> <C-w>l

"disable movement with arrow keys
map <up> <nop>
map <down> <nop>
map <left> <nop>
map <right> <nop>

imap <up> <nop>
imap <down> <nop>
imap <left> <nop>
imap <right> <nop>

"key mapping for tab navigation
nmap <Tab> gt
nmap <S-Tab> gT

"make <c-l> clear the highlight as well as redraw
nnoremap <C-c> :nohls<CR><C-L>
inoremap <C-c> <C-O>:nohls<CR>

" Switch to command mode without reaching up
inoremap kj <Esc>

""" Auto commands

"remove trailing whitespace when document is saved
autocmd BufWritePre * :%s/\s\+$//e

" Thorfile, Rakefile, Vagrantfile, Thorfile, Guardfile and Gemfile are Ruby
au BufRead,BufNewFile {Gemfile,Rakefile,Vagrantfile,Thorfile,Guardfile,config.ru}    set ft=ruby
au BufRead,BufNewFile *.{clj,cljs}    set ft=clojure

" md, markdown, and mk are markdown and define buffer-local preview
" au BufNewFile,BufRead *.{md,markdown,mdown,mkd,mkdn} call s:setupMarkup()
au BufNewFile,BufRead *.{md,markdown,mdown,mkd,mkdn} set ft=markdown

" add json syntax highlighting
au BufNewFile,BufRead *.json set ft=javascript

"au BufNewFile * set rnu

function! s:setupWrapping()
  set wrap
  set wrapmargin=2
  set textwidth=72
endfunction

au BufRead,BufNewFile *.txt call s:setupWrapping()

let mapleader=","

" Color scheme and font
if has('gui_running')
  colorscheme molokai
  set guifont=DejaVu\ Sans\ Mono:h13
else
  set background=dark
  colorscheme solarized
endif

" Directories for swp files
set backupdir=~/.vim/backup
set directory=~/.vim/backup

" To make vim copy-cut commands work as Cmd+C,Cmd+X
set clipboard=unnamed

" % to bounce from do to end etc.
runtime! macros/matchit.vim

" Detect changed files and offer to reload them
set autoread

" Copy (Yank) the file path of the current buffer
noremap <silent> <F4> :let @+=expand("%")<CR>

""""" Plugin configuration

" Ack and Ag
noremap <leader>a  :Ag!<space>
noremap <leader>A  :AckFromSearch<cr>
noremap <leader>wa :Ag!<space>-w<space>

" GitGrep
noremap gr :GitGrep<space>

" ZoomWin configuration
map <leader><leader> :ZoomWin<CR>

" NERDTree configuration
let NERDTreeChDirMode = 2
"let NERDTreeIgnore = ['\.pyc$', '\.rbc$', '\.git$', '\~$', '^tmp$', '^log$', '\.bundle$', '\.sass-cache', '\.swp$', '\.test-app-root$']
let NERDTreeQuitOnOpen = 0
let NERDTreeShowHidden = 1
let NERDTreeIgnore=['\~$', '^vendor$']
nnoremap <silent> <leader>n :NERDTreeToggle<cr>
nnoremap <silent> <C-f> :NERDTreeFind<cr>

"map to bufexplorer
nnoremap <leader>b :BufExplorer<cr>

" CtrlP
let g:ctrlp_map = '<leader>t'
let g:ctrlp_cmd = 'CtrlPMixed'
let g:ctrlp_custom_ignore = {
  \ 'dir': '\v[\/](tmp|node_modules|build|tmp|transpiled)$'
  \ }

" SplitJoin
nmap sj :SplitjoinJoin<cr>
nmap ss :SplitjoinSplit<cr>

" vim-mustache-handlebars
let g:mustache_abbreviations = 1

"jump to last cursor position when opening a file
"dont do it when writing a commit log entry
autocmd BufReadPost * call SetCursorPosition()
function! SetCursorPosition()
    if &filetype !~ 'commit\c'
        if line("'\"") > 0 && line("'\"") <= line("$")
            exe "normal! g`\""
            normal! zz
        endif
    end
endfunction

nmap <leader>f :put =expand('%:p')<CR>
```

I guess that makes it harder to isolate what's going wrong with CtrlP :) However, it does work perfectly with vim.

**Comment by balinterdi** [comment]
And now it's working fine, although really nothing (or more precisely nothing that I know of) has changed since yesterday 😮  . Anyway, let's keep this open for a few more days just in case the bug comes back.

**Comment by nicolaiskogheim** [comment]
> What is your **minimal** init.vim that makes the segfault reproducible for you?

@balinterdi There is a much greater chance that this issue gets traction if it's narrowed down as much as possible. You are the only one AFAICT that has been able to produce the segfault you mention, so people like me (casual "watchers" of this repository) generally wait until the issue is a litte bit more focused before trying to reproduce - because now it seems like the cause could originate from a couple of places even though it only manifested in CtrlP, and not many casuals like me would try to dissect your `.vimrc` at this point because if we can't reproduce the bug, we don't necessarily know that the bug doesn't manifest at our side, only that we weren't able to find it.

Since @oni-link wasn't able to reproduce with only the CtrlP plugin in the config, we know that there must be something else to this bug.

If the bug manifests again, try to narrow down your `.vimrc` as much as possible, and then people will be much more likely to try and help you find the cause 😄

**Comment by balinterdi** [comment]
Nicolai, you are totally right.

I felt kind of bad when I realized I did not do my due diligence by trying to narrow down the possible cause of the bug. I'll make sure to do that in the future if the bug rears its head again. 

It's also great to see that people are very responsive to issues here.

**Comment by maximsch2** [comment]
@balinterdi Are you using CtrlP with cpsm? Have you updated it recently? I think I hit an issue with cpsm building a wrong python module and messing things up.

**Comment by balinterdi** [comment]
Hey Maxim. Thank you for your comment. No, I don't have `cpsm` installed just `CtrlP`. However, it seems to work now, although the problem might creep back as I did not do anything that could have fixed it.

**Comment by justinmk** [comment]
Several crash-related fixes since 0.1.4, please try 0.1.6. Since there is no backtrace here we can't do much with this. If it happens again, on macOS/OSX it's very easy to [get a backtrace](https://github.com/neovim/neovim/wiki/Development-tips#backtrace-macos--osx), please do post the backtrace then. Thanks!

**Comment by balinterdi** [comment]
Hey Justin, thank you very much for your efforts. I upgraded to 0.1.6 about a week ago and has been sailing smoothly since then. Loving neovim more by the day!
